src/sharedHooks/useWatchPosition.ts

```diff
@@ -0,0 +1,125 @@
+import Geolocation, {
+  GeolocationError,
+  GeolocationResponse
+} from "@react-native-community/geolocation";
+import _ from "lodash";
+import { useEffect, useState } from "react";
+import {
+  RESULTS as PERMISSION_RESULTS
+} from "react-native-permissions";
+import {
+  checkLocationPermission,
+  shouldFetchObservationLocation,
+  TARGET_POSITIONAL_ACCURACY
+} from "sharedHelpers/shouldFetchObservationLocation.ts";
+import useStore from "stores/useStore";
+
+interface UserLocation {
+  latitude: number,
+  longitude: number,
+  positional_accuracy: number
+}
+
+interface WatchPosition {
+  hasLocation: boolean,
+  isFetchingLocation: boolean,
+  locationPermissionNeeded: boolean,
+  userLocation: UserLocation
+}
+
+const geolocationOptions = {
+  distanceFilter: 0,
+  enableHighAccuracy: true,
+  maximumAge: 0,
+  timeout: 2000
+};
+
+const useWatchPosition = ( options: {
+  retry: boolean
+} ): WatchPosition => {
+  const updateObservationKeys = useStore( state => state.updateObservationKeys );
+  const currentObservation = useStore( state => state.currentObservation );
+  const [currentPosition, setCurrentPosition] = useState<GeolocationResponse | null>( null );
+  const [subscriptionId, setSubscriptionId] = useState<number | null>( null );
+  const [locationPermissionResult, setLocationPermissionResult] = useState<string | null>( null );
+  const [errorCode, setErrorCode] = useState<number | null>( null );
+  const hasLocation = currentObservation?.latitude && currentObservation?.longitude;
+  const [shouldFetchLocation, setShouldFetchLocation] = useState<boolean>( true );
+  const [userLocation, setUserLocation] = useState<UserLocation | null>( null );
+
+  const watchPosition = ( ) => {
+    const success = ( position: GeolocationResponse ) => {
+      setLocationPermissionResult( "granted" );
```

Comment:
Elsewhere you configured `@react-native-community/geolocation` to `skipPermissionRequests` on the assumption that we will handle that... so there's no real reason to do this unless you really want to change `locationPermissionResult` to a new string object every single time we get a new a new location. I think we can just set the default value of `locationPermissionResult` to "granted" and only change it in the failure callback.